src/Functions/minSampleSize.cpp

```diff
@@ -0,0 +1,372 @@
+                Float64 sigma;
+                /// Minimal Detectable Effect
+                Float64 mde = col_mde.getFloat64(row_num);
+                /// Sufficient statistical power to detect a treatment effect
+                Float64 power = col_power.getFloat64(row_num);
+                /// Significance level
+                Float64 alpha = col_alpha.getFloat64(row_num);
+
+                if constexpr (op_case == OpCase::ConstBaseline)
+                {
+                    baseline = static_cast<Float64>(col_baseline[0]);
+                    sigma = static_cast<Float64>(col_sigma[row_num]);
+                }
+                else if constexpr (op_case == OpCase::ConstSigma)
+                {
+                    baseline = static_cast<Float64>(col_baseline[row_num]);
+                    sigma = static_cast<Float64>(col_sigma[0]);
+                }
+                else if constexpr (op_case == OpCase::Vector)
+                {
+                    baseline = static_cast<Float64>(col_baseline[row_num]);
+                    sigma = static_cast<Float64>(col_sigma[row_num]);
+                }
+                else if constexpr (op_case == OpCase::ConstBoth)
+                {
+                    baseline = static_cast<Float64>(col_baseline[0]);
+                    sigma = static_cast<Float64>(col_sigma[0]);
+                }
+
+                if (!std::isfinite(baseline) || !std::isfinite(sigma) || !isBetweenZeroAndOne(mde) || !isBetweenZeroAndOne(power) || !isBetweenZeroAndOne(alpha))
+                {
+                    data_min_sample_size.emplace_back(std::numeric_limits<Float64>::quiet_NaN());
+                    data_detect_lower.emplace_back(std::numeric_limits<Float64>::quiet_NaN());
+                    data_detect_upper.emplace_back(std::numeric_limits<Float64>::quiet_NaN());
+                    continue;
+                }
+
+                Float64 delta = baseline * mde;
+
+                using namespace boost::math;
+                normal_distribution<> nd(0.0, 1.0);
+                Float64 min_sample_size = 2 * (std::pow(sigma, 2)) * std::pow(quantile(nd, 1.0 - alpha / 2) + quantile(nd, power), 2) / std::pow(delta, 2);
+
+                data_min_sample_size.emplace_back(min_sample_size);
+                data_detect_lower.emplace_back(baseline - delta);
+                data_detect_upper.emplace_back(baseline + delta);
+            }
+
+            return ColumnTuple::create(Columns{std::move(res_min_sample_size), std::move(res_detect_lower), std::move(res_detect_upper)});
+        }
+
+        static ColumnPtr execute(const ColumnsWithTypeAndName & arguments, size_t input_rows_count)
+        {
+            const auto & arg_baseline = arguments[0];
+            const auto & arg_sigma = arguments[1];
+            ColumnPtr res;
+
+            const bool valid = castBothTypes(arg_baseline.type.get(), arg_sigma.type.get(), [&](const auto & left, const auto & right)
```

Comment:
@achimbab not sure, why this is needed here ? It seems that we have fixed types for these functions, or not ?

Reply:
@kitaisreal I removed this codes.